### PR TITLE
Import wallet srp password validation

### PR DESCRIPTION
**Pull request by chrisleewilcox**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

**Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the before and after of your change_

**Issue**

Progresses #???

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: wdio/features/step-definitions/onboarding.js [hunk 1]
```diff
@@ -100,22 +102,20 @@ When(/^I type (.*) in confirm password field/, async (text) => {
...
-      break;
-    case 'a valid Confirm password':
-      await ImportFromSeedScreen.typeConfirmPassword(validAccount.password);
-      break;
-    default:
-      throw new Error('Condition not found');
-  }
-})
+Then(/^device alert (.*) is displayed/, async (text) => {
+  const msg = await driver.getAlertText();
```
Comment (cortisiko) [inline wdio/features/step-definitions/onboarding.js:106]: line 106 to line 109 should be an assertion method that lives in the ImportWallet page object.

File: wdio/features/step-definitions/onboarding.js [hunk 2]
```diff
@@ -41,6 +42,7 @@ Then(/^"([^"]*)?" is displayed/, async (text) => {
       await ImportFromSeedScreen.verifyScreenTitle();
       break;
     case 'Welcome to your new wallet!':
+      // await driver.pause(10000);
```
Comment (cortisiko) [inline wdio/features/step-definitions/onboarding.js:45]: lets not leave comments in the code unless absolutely necessary.

File: app/components/Views/ImportFromSeed/index.js [hunk 3]
```diff
@@ -636,11 +638,11 @@ class ImportFromSeed extends PureComponent {
...
-                // {...generateTestId(
-                //   Platform,
-                //   IMPORT_FROM_SEED_SCREEN_NEW_PASSWORD_INPUT_ID,
-                // )}
-                testID={'create-password-first-input-field'}
+                {...generateTestId(
+                  Platform,
+                  IMPORT_FROM_SEED_SCREEN_NEW_PASSWORD_INPUT_ID,
+                )}
+                //testID={'create-password-first-input-field'}
```
Comment (cortisiko) [inline app/components/Views/ImportFromSeed/index.js:645]: let's not leave comments in the code

File: wdio/features/screen-objects/WalletMainScreen.js [hunk 4]
```diff
@@ -3,7 +3,7 @@ import { ONBOARDING_WIZARD_STEP_1_CONTAINER_ID } from '../testIDs/Components/Onb
 class WalletMainScreen {
   async validateOnboardingWizard() {
     const elem = await $(`~${ONBOARDING_WIZARD_STEP_1_CONTAINER_ID}`);
-    expect(elem).toBeDisplayed();
+    await expect(elem).toBeDisplayed();
```
Comment (cortisiko) [inline wdio/features/screen-objects/WalletMainScreen.js:6]: Create a getter for the object then assert that the object is visible. Something like:


```suggestion

  get onBoardingWizard() {
    return Selectors.getElementByPlatform(ONBOARDING_WIZARD_STEP_1_CONTAINER_ID);
  }

  async isOnboardingWizardVisible() {
    await expect(this.onBoardingWizard).toBeDisplayed();
  }
```

File: wdio/features/screen-objects/ImportFromSeedScreen.js [hunk 5]
```diff
@@ -50,6 +56,13 @@ class ImportFromSeed {
     await Gestures.waitAndTap(this.screenTitle);
     await Gestures.waitAndTap(this.importButton);
   }
+
+  async assertPasswordStrength(text){
+    const elem = await this.passwordStrength;
+    const strength = await elem.getText();
+    console.log(`## Password strength is: ${strength}`);
```
Comment (cortisiko) [inline wdio/features/screen-objects/ImportFromSeedScreen.js:63]: Was this for debugging purposes? Do not keep console log messages in the code.

File: wdio/features/screen-objects/ImportFromSeedScreen.js [hunk 6]
```diff
@@ -50,6 +56,13 @@ class ImportFromSeed {
     await Gestures.waitAndTap(this.screenTitle);
     await Gestures.waitAndTap(this.importButton);
   }
+
+  async assertPasswordStrength(text){
+    const elem = await this.passwordStrength;
```
Comment (cortisiko) [inline wdio/features/screen-objects/ImportFromSeedScreen.js:61]: this method seems like it is doing a lot. You are creating a variable for the password element then creating another variable to get the string of the password element. Why not do:

`const passwordStrengthText = await this.passwordStrength.getText();`

File: wdio/features/screen-objects/ImportFromSeedScreen.js [hunk 7]
```diff
@@ -50,6 +56,13 @@ class ImportFromSeed {
...
+    const strength = await elem.getText();
+    console.log(`## Password strength is: ${strength}`);
+    assert.strictEqual(strength, text);
```
Comment (cortisiko) [inline wdio/features/screen-objects/ImportFromSeedScreen.js:64]: any reason why we are not using webdriverio built-in assertions?

**Review by cortisiko — CHANGES_REQUESTED** [review]
see comments

File: app/components/Views/ImportFromSeed/index.js (hunk 3 above)
Comment (chrisleewilcox) [reply app/components/Views/ImportFromSeed/index.js:645]: This is a detox id that is commented out.  Are you sure you want this removed?

File: wdio/features/screen-objects/ImportFromSeedScreen.js (hunk 6 above)
Comment (chrisleewilcox) [reply wdio/features/screen-objects/ImportFromSeedScreen.js:61]: Done.

File: wdio/features/screen-objects/ImportFromSeedScreen.js (hunk 5 above)
Comment (chrisleewilcox) [reply wdio/features/screen-objects/ImportFromSeedScreen.js:63]: Removed.

File: wdio/features/screen-objects/ImportFromSeedScreen.js (hunk 7 above)
Comment (chrisleewilcox) [reply wdio/features/screen-objects/ImportFromSeedScreen.js:64]: Any reason why we want to only use webdriverio assertions?  No need for us to be limited with assertion capabilities.

File: wdio/features/screen-objects/WalletMainScreen.js (hunk 4 above)
Comment (chrisleewilcox) [reply wdio/features/screen-objects/WalletMainScreen.js:6]: This file was implemented by either Yande or Fatia.  I just made a simple fix by adding await in the return.  I would actually like this file to be renamed to "TakeATourComponent.js" or something like that and probably best to do maintenance work on another PR.

File: wdio/features/step-definitions/onboarding.js (hunk 2 above)
Comment (chrisleewilcox) [reply wdio/features/step-definitions/onboarding.js:45]: Removed.

File: wdio/features/step-definitions/onboarding.js (hunk 1 above)
Comment (chrisleewilcox) [reply wdio/features/step-definitions/onboarding.js:106]: Done.

File: app/components/Views/ImportFromSeed/index.js (hunk 3 above)
Comment (cortisiko) [reply app/components/Views/ImportFromSeed/index.js:645]: Is there a reason why you commented it out?

File: wdio/features/screen-objects/WalletMainScreen.js (hunk 4 above)
Comment (cortisiko) [reply wdio/features/screen-objects/WalletMainScreen.js:6]: That is fair.

File: wdio/features/screen-objects/ImportFromSeedScreen.js (hunk 6 above)
Comment (cortisiko) [reply wdio/features/screen-objects/ImportFromSeedScreen.js:61]: mmmm okay! Your function still does a lot for an assertion. 

how about splitting that function into two? One that gets the password strength text and the other for the assertion.

Comment (cortisiko) [reply wdio/features/screen-objects/ImportFromSeedScreen.js:61]:
```
async PasswordStrengthText(){     
    const passwordStrengthTextElement = await this.passwordStrength;     
    const strength = await passwordStrengthTextElement.getText();      
    return strength;   
  }  
```

```
  async assertPasswordStrength(text){     
    assert.strictEqual(await this.PasswordStrengthText(), text);       
    }
```

The assertion function does what it is set to do.

File: wdio/features/screen-objects/ImportFromSeedScreen.js [hunk 8]
```diff
@@ -50,6 +56,18 @@ class ImportFromSeed {
...
     await Gestures.waitAndTap(this.importButton);
   }
+
+  async assertPasswordStrength(text){
+    const elem = await this.passwordStrength;
+    const passwordStrengthText = await elem.getText();
+    assert.strictEqual(passwordStrengthText, text);      
+  }
+
+  async assertDeviceAlert(text){
```
Comment (cortisiko) [inline wdio/features/screen-objects/ImportFromSeedScreen.js:66]: This function should be split up into two. The action "tapping(accepting)" the alert should be in its function.

File: wdio/features/screen-objects/ImportFromSeedScreen.js (hunk 7 above)
Comment (cortisiko) [reply wdio/features/screen-objects/ImportFromSeedScreen.js]: I recommend using wdio assertions. In this case [toHaveText](https://webdriver.io/docs/api/expect-webdriverio/#tohavetext). I do agree with not being limited by assertions but I also believe it is important to have consistency. The question would be: when to use assert vs wdio assertions.

File: app/components/Views/ImportFromSeed/index.js (hunk 3 above)
Comment (chrisleewilcox) [reply app/components/Views/ImportFromSeed/index.js:645]: If I recall I think it was commented out when I created the branch.  I uncommented and doesn't look like that causes any issue.  Done.

File: wdio/features/screen-objects/ImportFromSeedScreen.js (hunk 7 above)
Comment (chrisleewilcox) [reply wdio/features/screen-objects/ImportFromSeedScreen.js]: This ended up being just a oneliner change that inherently getText() and compares so I didn't have to break that out into multiple functions.  

`await expect(this.passwordStrengthLabel).toHaveText(text);`

I like this approach.  Done.

File: wdio/features/screen-objects/ImportFromSeedScreen.js (hunk 8 above)
Comment (chrisleewilcox) [reply wdio/features/screen-objects/ImportFromSeedScreen.js:66]: Do we want to have a separate class like `/helpers/Device.js` and put device control functions there?  Can we discuss and address in another PR?

File: wdio/features/screen-objects/ImportFromSeedScreen.js (hunk 6 above)
Comment (chrisleewilcox) [reply wdio/features/screen-objects/ImportFromSeedScreen.js:61]: Done.  Related to below change.

File: wdio/features/screen-objects/ImportFromSeedScreen.js (hunk 8 above)
Comment (cortisiko) [reply wdio/features/screen-objects/ImportFromSeedScreen.js]: Yeah but you do so in this PR. This way, we lay a good foundation to deal with alert boxes when writing test cases in the future. It is as simple as creating a new class that has a function that taps "Ok" in the alert message.

Comment (chrisleewilcox) [reply wdio/features/screen-objects/ImportFromSeedScreen.js]: So I went down the route of trying to click "OK" on an alert and it didn't work out.  Doesn't look like there is a way to get an element for a device alert.  I then resorted to using webdriverio support for device alerts which is limited to...
[dismissAlert](https://webdriver.io/docs/api/jsonwp/#dismissalert)
[acceptAlert](https://webdriver.io/docs/api/jsonwp/#acceptalert)
[getAlertText](https://webdriver.io/docs/api/jsonwp/#getalerttext)
[sendAlertText](https://webdriver.io/docs/api/jsonwp/#sendalerttext)

Sounds like we need a class for device alert support for all our test.  There will also need to be some thought put into how we bundle up buttons on a device for acceptAlert(Yes, OK, Agree, Allow, etc...) and dismissAlert(No, Cancel, Close, etc...).  With all these considerations and others, I think this is separate work.  I feel more comfortable if there are no further changes necessary at this point related to SRP and password strength to merge this PR in before adding device support.  We should be mindful of not letting code reviews force PR's to drag on and take on other work that may be better suited in a separate PR.  Adding device support to this PR is essentially code creep for automation.  I agree we need it but think it's best suited in another PR so we can move on from this one.
Let me know how you want to proceed.

Comment (cortisiko) [reply wdio/features/screen-objects/ImportFromSeedScreen.js]: Thank you for doing the necessary research. The purpose of the code review is to ensure we are making the code as clean and readable as possible not necessarily dragging on the PR.

Comment (cortisiko) [reply wdio/features/screen-objects/ImportFromSeedScreen.js]: I will ask Fatia or Yande to handle. Thanks!

**Review by cortisiko — APPROVED** [review]
Much better. Feel free to merge whenever @chrisleewilcox

**Pull request merged**